src/envs/pybullet_blocks.py

```diff
@@ -0,0 +1,728 @@
+                    ("rel", "rel", "abs"), types, params_space),
+            ])
+
+        # One-time initialization of pybullet assets. Note that this happens
+        # in __init__ because many class attributes are created.
+        # Skip test coverage because GUI is too expensive to use in unit tests
+        # and cannot be used in headless mode.
+        if CFG.pybullet_use_gui:  # pragma: no cover
+            self._physics_client_id = p.connect(p.GUI)
+            p.resetDebugVisualizerCamera(
+                self._camera_distance,
+                self._camera_yaw,
+                self._camera_pitch,
+                self._camera_target,
+                physicsClientId=self._physics_client_id)
+        else:
+            self._physics_client_id = p.connect(p.DIRECT)
+
+        p.resetSimulation(physicsClientId=self._physics_client_id)
+
+        # Load plane.
+        p.loadURDF(utils.get_env_asset_path("urdf/plane.urdf"), [0, 0, -1],
+                   useFixedBase=True,
+                   physicsClientId=self._physics_client_id)
+
+        # Load Fetch robot.
+        self._fetch_id = p.loadURDF(
+            utils.get_env_asset_path("urdf/robots/fetch.urdf"),
+            useFixedBase=True,
+            physicsClientId=self._physics_client_id)
+        p.resetBasePositionAndOrientation(
+            self._fetch_id,
+            self._base_pose,
+            self._base_orientation,
+            physicsClientId=self._physics_client_id)
+
+        # Extract IDs for individual robot links and joints.
+        joint_names = [
+            p.getJointInfo(
+                self._fetch_id, i,
+                physicsClientId=self._physics_client_id)[1].decode("utf-8")
+            for i in range(
+                p.getNumJoints(self._fetch_id,
+                               physicsClientId=self._physics_client_id))
+        ]
+        self._ee_id = joint_names.index('gripper_axis')
+        self._arm_joints = get_kinematic_chain(
+            self._fetch_id,
+            self._ee_id,
+            physics_client_id=self._physics_client_id)
+        self._left_finger_id = joint_names.index("l_gripper_finger_joint")
+        self._right_finger_id = joint_names.index("r_gripper_finger_joint")
+        self._arm_joints.append(self._left_finger_id)
+        self._arm_joints.append(self._right_finger_id)
+
+        # Load table.
+        self._table_id = p.loadURDF(
+            utils.get_env_asset_path("urdf/table.urdf"),
+            useFixedBase=True,
+            physicsClientId=self._physics_client_id)
+        p.resetBasePositionAndOrientation(
+            self._table_id,
+            self._table_pose,
+            self._table_orientation,
+            physicsClientId=self._physics_client_id)
+
+        # Skip test coverage because GUI is too expensive to use in unit tests
+        # and cannot be used in headless mode.
+        if CFG.pybullet_draw_debug:  # pragma: no cover
```

Comment:
I tried to do this, but it doesn't work because the drawing needs to happen after resetSimulation. but yes, this can only happen if the GUI is on. I'll add an assert